supplementary/alica_ros1/supplementary_tests/libalica-supplementary-tests/include/libalica-supplementary-tests/GSolver/GSolverTestPlan.h

```diff
@@ -23,15 +23,15 @@ class GSolverTestPlanUtilityFunction : public BasicUtilityFunction
 {
 public:
     std::shared_ptr<UtilityFunction> getUtilityFunction(Plan* plan);
-    static std::unique_ptr<GSolverTestPlanUtilityFunction> create(alica::UtilityFunctionContext& context);
+    static std::shared_ptr<GSolverTestPlanUtilityFunction> create(alica::UtilityFunctionContext& context);
```

Comment:
does anyone know how to verify the type of a loaded symbol? The symbol is loaded using boost::import_alias which does not seem to verify this & the program can act in weird ways if the types don't match. Somehow, these errors were not caught earlier as returning unique_ptr instead of shared_ptr did not cause any crashes up until this PR

Reply:
IIUC this problem, it related to c++ types system. Return type is not a part of function type.

Reply:
yes, but we need someway I think to ensure that we can detect if the right type is returned by the creator in the engine, otherwise the program can crash or do unexpected things